olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -517,8 +518,37 @@ bool VersionedLayerClientImpl::RemoveFromCache(
 }
 
 bool VersionedLayerClientImpl::RemoveFromCache(const geo::TileKey& tile) {
-  auto partition_id = tile.ToHereTile();
-  return RemoveFromCache(partition_id);
+  read::QuadTreeIndex cached_tree;
+  if (repository::PartitionsRepository::FindQuadTree(
+          catalog_, settings_, layer_id_, catalog_version_.load(), tile,
+          cached_tree)) {
+    auto data = cached_tree.Find(tile, false);
+    if (!data) {
+      return true;
+    }
+    repository::DataCacheRepository cache_repository(catalog_, settings_.cache);
+    auto result = cache_repository.Clear(layer_id_, data->data_handle);
+    if (result) {
+      auto need_remove = true;
+      auto index_data = cached_tree.GetIndexData();
+      for (const auto& ind : index_data) {
+        if (ind.tile_key != tile &&
+            cache_repository.IsCached(layer_id_, ind.data_handle)) {
+          need_remove = false;
+          break;
```

Comment:
```suggestion
          return true;
```
Our job is done here.